src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs

```diff
@@ -940,7 +961,7 @@ private void CompileComplexContentExtension(XmlSchemaComplexType complexType, Xm
                 }
             }
             complexType.SetBaseSchemaType(baseType);
-            if ((baseType.FinalResolved & XmlSchemaDerivationMethod.Extension) != 0)
+            if ((baseType!.FinalResolved & XmlSchemaDerivationMethod.Extension) != 0)
```

Comment:
I also suspect it's useless check but didn't feel as confident here to remove it here since this code is relatively complex and code is internal anyway and won't affect public APIs. I'll leave this as is